spec/unit/provider/package/rpm_spec.rb

```diff
@@ -123,6 +123,13 @@
         provider.upgrade_package("ImageMagick-c++", "6.5.4.7-7.el6_5")
       end
 
+      it "should run rpm -U --oldpackage with the package source to downgrade" do
+        allow(new_resource).to receive(:allow_downgrade).and_return(true)
```

Comment:
I'd prefer it changed, that pattern has caused bugs in other provider code.  You're changing the way the `new_resource` responds to external requests here, but internally there is still a `@allow_downgrade` ivar as internal state which you haven't changed.  It doesn't matter in this test, but you could poke the `new_resource` via a different method that read `@allow_downgrade` directly and it'd surprise you.  If someone adds behavior like that later, then your code plus their code could wind up being a test bug.